test/common/TestCommon.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE DataKinds, RankNTypes, TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+module TestCommon ((+\), (*\), (**\),
+                   listsToParameters,
+                   cmpTwo, cmpTwoSimple,
+                   qcPropDom, quickCheckTest0, fquad, quad
+                  ) where
+
+import Prelude
+
+import qualified Data.Array.DynamicS as OT
+import qualified Data.Array.ShapedS as OS
+import qualified Data.Vector.Generic as V
+import           Numeric.LinearAlgebra (Vector)
+import qualified Numeric.LinearAlgebra as HM
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+import HordeAd hiding (sumElementsVectorOfDual)
+
+(+\) :: DualMonad d r m
+     => DualNumber d r -> DualNumber d r -> m (DualNumber d r)
+(+\) u v = returnLet $ u + v
+
+(*\) :: DualMonad d r m
+     => DualNumber d r -> DualNumber d r -> m (DualNumber d r)
+(*\) u v = returnLet $ u * v
+
+(**\) :: DualMonad d r m
+      => DualNumber d r -> DualNumber d r -> m (DualNumber d r)
+(**\) u v = returnLet $ u ** v
+
+-- Checks if 2 numbers are close enough.
+close1 :: Double -> Double -> Bool
```

Comment:
Since we are generalizing, could we generalize this to arbitrary `Num r`? And similarly everywhere else in this file. We need a concrete type as soon as we actually test, but in test tools, restriction to Double only adds noise.